src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AzureRMExpressRouteCrossConnectionPeeringBase.cs

```diff
@@ -0,0 +1,137 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Network
+{
+    using System.Collections.Generic;
+    using System.ComponentModel;
+
+    using Microsoft.Azure.Commands.Network.Models;
+    using System.Management.Automation;
+    using MNM = Microsoft.Azure.Management.Network.Models;
+    using System.Linq;
+
+    public class AzureRMExpressRouteCrossConnectionPeeringBase : NetworkBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The name of the Peering")]
+        [ValidateNotNullOrEmpty]
+        public virtual string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The PeeringType")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(
+           MNM.ExpressRoutePeeringType.AzurePrivatePeering,
+           MNM.ExpressRoutePeeringType.AzurePublicPeering,
+           MNM.ExpressRoutePeeringType.MicrosoftPeering,
+           IgnoreCase = true)]
+        public string PeeringType { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The PeerAsn")]
+        [ValidateNotNullOrEmpty]
+        public uint PeerASN { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The PrimaryPeerAddressPrefix")]
+        [ValidateNotNullOrEmpty]
+        public string PrimaryPeerAddressPrefix { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The SecondaryPeerAddressPrefix")]
+        [ValidateNotNullOrEmpty]
+        public string SecondaryPeerAddressPrefix { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The vlanId")]
+        [ValidateNotNullOrEmpty]
+        public int VlanId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The SharedKey")]
+        [ValidateNotNullOrEmpty]
+        public string SharedKey { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The MircosoftConfigAdvertisedPublicPrefixes")]
+        [ValidateNotNullOrEmpty]
+        public List<string> MicrosoftConfigAdvertisedPublicPrefixes { get; set; }
```

Comment:
This should be singular `MicrosoftConfigAdvertisedPublicPrefix` and use type `string[]`.

Reply:
Need to keep this consistent with the BGP peering cmdlets under circuit

Reply:
We don't allow grandfathered logic from prior cmdlets to break PowerShell design guidelines unless there is business justification to do so. The BGP cmdlets were probably made prior to us enforcing PowerShell design practices. What is recommended is adhering to the design guidelines for new cmdlets and then updating your old cmdlets in the next breaking change release.

Reply:
Fixed